### PR TITLE
Tagging2d

### PR DESCRIPTION
**Pull request by nicolasaunai**

### CODE REVIEW

File: src/amr/tagging/default_hybrid_tagger_strategy.h [hunk 1]
```diff
@@ -28,49 +29,159 @@ void DefaultHybridTaggerStrategy<HybridModel>::tag(HybridModel& model,
 
     auto& N = model.state.ions.density();
 
+    double threshold = 0.1;
+
     // we loop on cell indexes for all qties regardless of their centering
-    auto start
+    auto start_x
```
Comment (PhilipDeegan) [inline src/amr/tagging/default_hybrid_tagger_strategy.h:35]: could be compacted with 

```
auto [start_x, end_x] = layout.physicalStartToEnd(PHARE::core::QtyCentering::dual, PHARE::core::Direction::X);
```

File: src/amr/tagging/default_hybrid_tagger_strategy.h [hunk 2]
```diff
@@ -28,49 +29,159 @@ void DefaultHybridTaggerStrategy<HybridModel>::tag(HybridModel& model,
...
+                tagsv(iCell) = 1;
             }
             else
-                tags[iCell] = 0;
+                tagsv(iCell) = 0;
+        }
+    }
+    if constexpr (dimension == 2)
+    {
+        auto start_y
```
Comment (PhilipDeegan) [inline src/amr/tagging/default_hybrid_tagger_strategy.h:77]: similarly

```
auto [start_y, end_y] = layout.physicalStartToEnd(PHARE::core::QtyCentering::dual, PHARE::core::Direction::Y);
```

File: src/amr/tagging/default_hybrid_tagger_strategy.h [hunk 3]
```diff
@@ -28,49 +29,159 @@ void DefaultHybridTaggerStrategy<HybridModel>::tag(HybridModel& model,
...
 
-    auto endCell     = layout.nbrCells()[0] - 1;
-    double threshold = 0.05;
-    int idx          = 4;
-    double epsilon   = 0.1;
+    auto endCell_x = layout.nbrCells()[0] - 1;
 
+    // SAMRAI tags int* buffer is FORTRAN ordering so we set false to the view
+    auto tagsv
+        = core::NdArrayView<dimension, int, int*, /*c_ordering=*/false>(tags, layout.nbrCells());
```
Comment (PhilipDeegan) [inline src/amr/tagging/default_hybrid_tagger_strategy.h:44]: we could probably throw a 

```
template<std::size_t dim, typename DataType>
using FortranNdArrayView = NdArrayView<dim, DataType, DataType*, /*c_ordering=*/false>;
````

into ndarray_vector.h somewhere

File: src/amr/tagging/default_hybrid_tagger_strategy.h [hunk 4]
```diff
@@ -28,49 +29,159 @@ void DefaultHybridTaggerStrategy<HybridModel>::tag(HybridModel& model,
...
+        auto end_y
+            = layout.physicalEndIndex(PHARE::core::QtyCentering::dual, PHARE::core::Direction::X);
+
+        auto endCell_y = layout.nbrCells()[1] - 1;
+
+        for (auto iTag_x = 0u, ix = start_x; iTag_x <= endCell_x; ++ix, ++iTag_x)
+        {
+            for (auto iTag_y = 0u, iy = start_y; iTag_y <= endCell_y; ++iy, ++iTag_y)
+            {
+                auto Bxavg_x = 0.2
```
Comment (PhilipDeegan) [inline src/amr/tagging/default_hybrid_tagger_strategy.h:88]: Are these identical apart from the input? (Bx/By/Bz/N)

```
                auto field_avg = [](auto const& F){
                  return std::make_tuple(
                          0.2
                               * (F(ix - 2, iy) + F(ix - 1, iy) + F(ix, iy) + F(ix + 1, iy)
                                  + F(ix + 2, iy)), 0.2
                               * (F(ix, iy - 2) + F(ix, iy - 1) + F(ix, iy) + F(ix, iy + 1)
                                  + F(ix, iy + 2)),  0.2
                                 * (F(ix - 1, iy) + F(ix, iy) + F(ix + 1, iy) + F(ix + 2, iy)
                                    + F(ix + 3, iy)), 0.2
                                 * (F(ix, iy - 1) + F(ix, iy) + F(ix, iy + 1) + F(ix, iy + 2)
                                    + F(ix, iy + 3))
                    );
                };

                auto const& [Bxavg_x, Bxavg_y, Bxavgp1_x, Bxavgp1_y] = field_avg(Bx);
                // By
                // Bz
                // N
```

File: src/amr/tagging/hybrid_tagger.h [hunk 5]
```diff
@@ -80,7 +81,24 @@ void HybridTagger<HybridModel>::tag(PHARE::solver::IPhysicalModel<amr_t>& model,
                 = std::make_shared<typename Map_value_type::element_type>(layout.nbrCells());
         }
 
-        std::copy(tags, tags + layout.nbrCellsFlat(), hybridModel.tags[key]->data());
+        //        std::copy(tags, tags + layout.nbrCellsFlat(), hybridModel.tags[key]->data());
```
Comment (PhilipDeegan) [inline src/amr/tagging/hybrid_tagger.h:84]: torm?

File: src/phare/phare.h [hunk 6]
```diff
@@ -32,7 +32,7 @@ class SamraiLifeCycle
         SAMRAI::tbox::SAMRAI_MPI::init(&argc, &argv);
         SAMRAI::tbox::SAMRAIManager::initialize();
         SAMRAI::tbox::SAMRAIManager::startup();
-
+        SAMRAI::tbox::SAMRAI_MPI::setCallAbortInParallelInsteadOfMPIAbort();
```
Comment (PhilipDeegan) [inline src/phare/phare.h:35]: not opposed

File: tests/simulator/test_tagging.py [hunk 7]
```diff
@@ -124,7 +124,7 @@ def ddt_test_id(self):
     @data(*_test_cases)
     def test_tagging(self, simInput):
         # UPDATE when 2d tagging is finished
-        for ndim in [1]: #supported_dimensions():
+        for ndim in [1,2]: #supported_dimensions():
```
Comment (PhilipDeegan) [inline tests/simulator/test_tagging.py:127]: could be "supported_dimensions()" again?

File: src/amr/tagging/default_hybrid_tagger_strategy.h (hunk 3 above)
Comment (PhilipDeegan) [reply src/amr/tagging/default_hybrid_tagger_strategy.h:44]: although maybe insufficient for ```DataType const*```

Comment (nicolasaunai) [reply src/amr/tagging/default_hybrid_tagger_strategy.h:44]: :shrug:  ?

**Review by PhilipDeegan — APPROVED** [review]

**Pull request closed without merging**